stl/inc/format

```diff
@@ -1758,7 +1767,19 @@ private:
     void _Flush() {
         auto _Size = this->_Size();
         this->_Clear();
-        _Output = _STD _Copy_unchecked(_Data, _Data + this->_Limit(_Size), _STD move(_Output));
+        const auto _End = _Data + this->_Limit(_Size);
+
+        if constexpr (_Is_specialization_v<_OutputIt, back_insert_iterator>) {
+            using _Container = typename _OutputIt::container_type;
+            if constexpr (_Is_specialization_v<_Container, basic_string> || _Is_specialization_v<_Container, vector>) {
+                auto& _Cont = *_Back_insert_iterator_container_access<_Container>{_Output}.container;
+                _Cont.insert(_Cont.end(), _Data, _End);
+            } else {
+                _Output = _STD _Copy_unchecked(_Data, _End, _STD move(_Output));
+            }
+        } else {
+            _Output = _STD _Copy_unchecked(_Data, _End, _STD move(_Output));
+        }

```

Comment:
This repeated line of code looks a bit odd. Can we get rid of it?
With something like that:
```cpp
if constexpr (conjunction_v<_Is_specialization<_OutputIt, back_insert_iterator>,
	disjunction<_Is_specialization<_OutputIt::container_type, basic_string>,
	_Is_specialization<_OutputIt::container_type, vector>>>) {
	auto& _Cont = *_Back_insert_iterator_container_access<_OutputIt::container_type>{_Output}.container;
	_Cont.insert(_Cont.end(), _Data, _End);
}
else {
	_Output = _STD _Copy_unchecked(_Data, _End, _STD move(_Output));
}
```

Reply:
This doesn't works, because the whole expression is parsed at once (not lazily evaluated/parsed like runtime boolean expressions), so we'll get `'container_type': is not a member of '_OutputIt'` errors even when `_Is_specialization<_OutputIt, back_insert_iterator>` gives false.

Reply:
According to [cppreference](https://en.cppreference.com/w/cpp/types/conjunction) you are wrong:

> Conjunction is short-circuiting: if there is a template type argument Bi with bool(Bi::value) == false, then instantiating conjunction<B1, ..., BN>::value does not require the instantiation of Bj::value for j > i.

Reply:
It still needs to be a valid expression before instantiation: https://godbolt.org/z/PWo3d4x5z

Reply:
You're right here.

That's unfortunate, but the *compile-time* territory is a wild beast.

Reply:
How about this variant then:
```cpp
void _Flush() {
	auto _Size = this->_Size();
	this->_Clear();
	const auto _End = _Data + this->_Limit(_Size);

	if constexpr (_Is_specialization_v<_OutputIt, back_insert_iterator>) {
		using _Container = typename _OutputIt::container_type;
		if constexpr (_Is_specialization_v<_Container, basic_string> || _Is_specialization_v<_Container, vector>) {
			auto& _Cont = *_Back_insert_iterator_container_access<_Container>{_Output}.container;
			_Cont.insert(_Cont.end(), _Data, _End);
			return;
		}
	}
	_Output = _STD _Copy_unchecked(_Data, _End, _STD move(_Output));
}
```
`else` after `return` is unnecessary.

Reply:
At `/W4`, this will give `warning C4702: unreachable code` and debug codegen will still emit the full `_Copy_unchecked` code for `back_insert_iterator<string>` (it gets correctly removed in release however). A type trait to extract `container_type` or void would solve both issues at once here, so I'll add that.

Reply:
> At `/W4`, this will give `warning C4702: unreachable code`

I don't get it in 16.10 in Release nor in Debug. I think it's the same as `warning c4100: unreferenced formal parameter`, so compiler should be able to figure out what's happening here.
But your type trait also looks good.